src/OptionSettlement.sol

```diff
@@ -188,31 +188,52 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
 
     /// @inheritdoc IOptionSettlementEngine
     function write(uint256 optionId, uint112 amount) external returns (uint256 claimId) {
-        if (amount == 0) {
-            revert AmountWrittenCannotBeZero();
-        }
+        /// claimId == 0 signifies that a new claim NFT should be minted
+        return write(0, optionId, amount);
+    }
 
+    /// @inheritdoc IOptionSettlementEngine
+    function write(uint256 claimId, uint256 optionId, uint112 amount) public returns (uint256) {
         (uint160 _optionId, uint96 claimIndex) = getDecodedIdComponents(optionId);
+
+        // claim index must be zero in lower 96b for provided option Id
         if (claimIndex != 0) {
-            // claim index must be zero in lower 96b
             revert InvalidOption(optionId);
         }
 
-        Option storage optionRecord = _option[_optionId];
-
-        if (optionRecord.expiryTimestamp <= block.timestamp) {
-            revert ExpiredOption(optionId, optionRecord.expiryTimestamp);
+        // claim provided must match the option provided
+        if (claimId != 0 && (claimId >> 96) != (optionId >> 96)) {
+            revert EncodedOptionIdInClaimIdDoesNotMatchProvidedOptionId(claimId, optionId);
         }
 
-        uint256 rxAmount = amount * optionRecord.underlyingAmount;
-        uint256 fee = ((rxAmount / 10000) * feeBps);
-        address underlyingAsset = optionRecord.underlyingAsset;
+        Option storage optionRecord = _writeOptions(_optionId, amount);
+        uint256 mintClaimNft = 0;
+
+        if (claimId == 0) {
```

Comment:
Again here, the assumtion about the zeroness of claimID, we want to support many claims per option type.

Reply:
I think the variable naming is what's confusing here. The claim ID is not assumed to be zero. Rather, we require that the lower 96b of the provided option ID are zero (perhaps mislabeled as `claimIndex`, as to be distinguisehd from the provided `claimId`). Here, if the claimID is zero, we create a new claim.